### PR TITLE
Add ignition-msgs

### PR DESCRIPTION
**Pull request by joxoby**
Specify library name and version:  **ignition-msgs/6.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by joxoby** [comment]
Depends on:
[protobuf](https://github.com/conan-io/conan-center-index/pull/2037)
[ignition-cmake](https://github.com/conan-io/conan-center-index/pull/3268)
[ignition-math](https://github.com/conan-io/conan-center-index/pull/3215)

File: recipes/ignition-msgs/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    @property
+    def _cmake_build_subfolder(self):
+        return "cmake_build_subfolder"
+
+    @property
+    def _cmake_prefix_subfolder(self):
+        return "cmake_prefix_subfolder"
+
+    def source(self):
```
Comment (madebr) [inline recipes/ignition-msgs/all/conanfile.py:51]:
```suggestion
        return "source_subfolder"

    def source(self):
```

File: recipes/ignition-msgs/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        self.build_requires("pkgconf/1.7.3")
+        self.build_requires("ignition-cmake/2.5.0")
+
+    def _configure_cmake(self):
+        if self._cmake is not None:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.verbose = True
+        self._cmake.definitions["BUILD_TESTING"] = False
+        # self._cmake.definitions["CMAKE_PREFIX_PATH"] = os.path.join(self.build_folder, self._cmake_prefix_subfolder)
```
Comment (madebr) [inline recipes/ignition-msgs/all/conanfile.py:100]:
```suggestion
```

File: recipes/ignition-msgs/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        # Remove MS runtime files
+        for dll_pattern_to_remove in ["concrt*.dll", "msvcp*.dll", "vcruntime*.dll"]:
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), dll_pattern_to_remove)
+
+    def package_info(self):
+        version_major = tools.Version(self.version).major
+        self.cpp_info.names["cmake_find_package"] = "ignition-msgs{}".format(version_major)
+        self.cpp_info.names["cmake_find_package_multi"] = "ignition-msgs{}".format(version_major)
+
+        # FIXME: create in file ignition-msgs6-all-config.cmake
```
Comment (madebr) [inline recipes/ignition-msgs/all/conanfile.py:127]: Have you manually checked that ignition-msgs creates only one `-config.cmake` file?
If so, then this FIXME can be removed, as the problem with `ignition-math` is that it needs to generate multiple `-config.cmake` files (which conan cannot do yet).

File: recipes/ignition-msgs/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.10.2)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(IGN_MSGS_MAJOR_VER "" CACHE STRING "Version of igition-math")
```
Comment (madebr) [inline recipes/ignition-msgs/all/test_package/CMakeLists.txt:7]:
```suggestion
set(IGN_MSGS_MAJOR_VER "" CACHE STRING "Version of ignition-msgs")
```

File: recipes/ignition-msgs/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        self.cpp_info.components["libignition-msgs-all"].libs = []
```
Comment (madebr) [inline recipes/ignition-msgs/all/conanfile.py:128]: Is this correct? No libs means that this package is header-only.

File: recipes/ignition-msgs/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        self.cpp_info.components["libignition-msgs-all"].requires = ["libignition-msgs-eigen3"]
```
Comment (madebr) [inline recipes/ignition-msgs/all/conanfile.py:129]: This should contain the conan component names on which this component (=`libignition-msgs-all`) depends.
In this case these are: `["tinyxml2::tinyxml2", "protobuf::protobuf", "ignition-math::ignition-math"]`.

**Pull request closed without merging**